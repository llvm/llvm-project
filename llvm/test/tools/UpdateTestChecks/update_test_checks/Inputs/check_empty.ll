; RUN: opt < %s -S | FileCheck %s

; Test whether the UTC check empty lines instead of skipping them.
define i32 @test(i32 %x) {
entry:
  br label %block1

block1:
  %cmp = icmp eq i32 %x, 0
  br i1 %cmp, label %block2, label %exit1

block2:
  br i1 %cmp, label %block3, label %exit2

block3:
  br i1 %cmp, label %exit3, label %exit4

exit1:
  ret i32 0

exit2:
  ret i32 %x

exit3:
  ret i32 %x

exit4:
  ret i32 %x
}
