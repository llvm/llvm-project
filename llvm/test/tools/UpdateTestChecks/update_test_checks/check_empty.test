## test whether the UTC generates CHECK-EMPTY for blank lines
# RUN: cp -f %S/Inputs/check_empty.ll %t.ll && %update_test_checks %t.ll --version 7
# RUN: diff -u %t.ll %S/Inputs/check_empty.ll.expected
