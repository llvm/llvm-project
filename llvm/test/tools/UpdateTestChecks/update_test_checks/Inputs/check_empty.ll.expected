; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 6
; RUN: opt < %s -S | FileCheck %s

; Test whether the UTC check empty lines instead of skipping them.
define i32 @test(i32 %x) {
; CHECK-LABEL: define i32 @test(
; CHECK-SAME: i32 [[X:%.*]]) {
; CHECK-NEXT:  [[ENTRY:.*:]]
; CHECK-NEXT:    br label %[[BLOCK1:.*]]
; CHECK-EMPTY:
; CHECK-NEXT:  [[BLOCK1]]:
; CHECK-NEXT:    [[CMP:%.*]] = icmp eq i32 [[X]], 0
; CHECK-NEXT:    br i1 [[CMP]], label %[[BLOCK2:.*]], label %[[EXIT1:.*]]
; CHECK-EMPTY:
; CHECK-NEXT:  [[BLOCK2]]:
; CHECK-NEXT:    br i1 [[CMP]], label %[[BLOCK3:.*]], label %[[EXIT2:.*]]
; CHECK-EMPTY:
; CHECK-NEXT:  [[BLOCK3]]:
; CHECK-NEXT:    br i1 [[CMP]], label %[[EXIT3:.*]], label %[[EXIT4:.*]]
; CHECK-EMPTY:
; CHECK-NEXT:  [[EXIT1]]:
; CHECK-NEXT:    ret i32 0
; CHECK-EMPTY:
; CHECK-NEXT:  [[EXIT2]]:
; CHECK-NEXT:    ret i32 [[X]]
; CHECK-EMPTY:
; CHECK-NEXT:  [[EXIT3]]:
; CHECK-NEXT:    ret i32 [[X]]
; CHECK-EMPTY:
; CHECK-NEXT:  [[EXIT4]]:
; CHECK-NEXT:    ret i32 [[X]]
;
entry:
  br label %block1

block1:
  %cmp = icmp eq i32 %x, 0
  br i1 %cmp, label %block2, label %exit1

block2:
  br i1 %cmp, label %block3, label %exit2

block3:
  br i1 %cmp, label %exit3, label %exit4

exit1:
  ret i32 0

exit2:
  ret i32 %x

exit3:
  ret i32 %x

exit4:
  ret i32 %x
}
